api/api_full.go

```diff
@@ -936,6 +957,39 @@ type FullNode interface {
 	// F3IsRunning returns true if the F3 instance is running, false if it's not running but
 	// it's enabled, and an error when disabled entirely.
 	F3IsRunning(ctx context.Context) (bool, error) //perm:read
+	// F3GetProgress returns the progress of the current F3 instance in terms of instance ID, round and phase.
+	F3GetProgress(ctx context.Context) (F3Progress, error) //perm:read
+}
+
+// F3ParticipationTicket represents a ticket that authorizes a miner to
+// participate in the F3 consensus.
+type F3ParticipationTicket []byte
+
+// F3ParticipationLease defines the lease granted to a storage provider for
+// participating in F3 consensus, detailing the session identifier, issuer,
+// subject, and the expiration instance.
+type F3ParticipationLease struct {
+	// Session uniquely identifies the session in which this lease is issued. This
+	// information can then be used to infer if the issuer of the ticket has restarted
+	// since the lease was issued.
+	Session uuid.UUID
```

Comment:
Do we need this to be `uuid.UUID` (it is a new dependency for API users)? Could it be just a string?

Reply:
> it is a new dependency for API users

`Common.Session` API already depends on it. 

For the purposes of this PR it's primarily for an informative log line when a node restarts. Otherwise has no effect on the leasing functionality. Happy to remove it and add later if we are not bothered about it.

Reply:
Removed Session. We can add later if there is a need for it beyond logging.